src/main/java/com/parkinglot/parkinglotfees/service/impl/ParkingLotBookingImpl.java

```diff
@@ -1,94 +1,160 @@
 package com.parkinglot.parkinglotfees.service.impl;
 
 import com.parkinglot.parkinglotfees.entity.*;
+import com.parkinglot.parkinglotfees.exception.ErrorCodes;
+import com.parkinglot.parkinglotfees.exception.ParkingServiceException;
 import com.parkinglot.parkinglotfees.repository.FeeSchemaRepository;
 import com.parkinglot.parkinglotfees.repository.OrganisationRepository;
 import com.parkinglot.parkinglotfees.repository.ParkingLotRepository;
 import com.parkinglot.parkinglotfees.repository.VehicleRepository;
 import com.parkinglot.parkinglotfees.service.ParkingLotBooking;
-import org.springframework.beans.factory.annotation.Autowired;
+import com.parkinglot.parkinglotfees.utils.ParkingUtils;
 import org.springframework.stereotype.Service;
 
-import javax.annotation.PostConstruct;
 import java.time.LocalDateTime;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
 import java.util.stream.Collectors;
+import java.util.stream.IntStream;
 
 @Service
 public class ParkingLotBookingImpl implements ParkingLotBooking {
-//    @Autowired
     private OrganisationRepository organisationRepository;
-    //private ModelMapper mapper;
-//    @Autowired
     private ParkingLotRepository parkingLotRepository;
 
-//    @Autowired
     private FeeSchemaRepository feeSchemaRepository;
     private VehicleRepository vehicleRepository;
 
-
     public ParkingLotBookingImpl(OrganisationRepository organisationRepository, ParkingLotRepository parkingLotRepository, FeeSchemaRepository feeSchemaRepository,
                                  VehicleRepository vehicleRepository) {
         this.organisationRepository = organisationRepository;
         this.parkingLotRepository = parkingLotRepository;
         this.feeSchemaRepository = feeSchemaRepository;
         this.vehicleRepository = vehicleRepository;
     }
-    @PostConstruct
-    public void getMyMethod(){
 
+    private Integer getVehicleCountByOrg(Organisation organisation, String vehicleType) {
+        if (vehicleType.equals(VehicleType.BUS.toString())) {
```

Comment:
please do the inverted string comparison to avoid NPE

Reply:
done